### ISSUE TITLE
Concentrate formatter changes in next release

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#13002]
We're frequently merging formatter changes, but usually it's tiny bug fixes for edge cases that don't affect much code.

With #12951 we have a formatter change in the development branch that will affect quite a lot of code bases.
I think it would be a good chance to introduce other impactful formatter changes in the same release. This would concentrate these changes in a single release and users don't need to constantly reformat their code bases. It's easier to do all changes A, B, C all in one go for release 1.8 instead of A in 1.8, B in 1.9 C in 1.10 etc.

These are some formatter changes pending implementation that I could find in the issue tracker:

* [ ] #12938
* [ ] Reformat unescaped Unicode bi-directional control characters (https://github.com/crystal-lang/crystal/pull/12923#issuecomment-1399342541)
* [ ] #11949
* [ ] #12999 
* [ ] #5074
* [ ] #9154
* [ ] #11209
* [ ] #11478 (experimental)
* [ ] #7272 (experimental)
* [ ] #9804 (experimental)

We probably won't be able to address all for the next release, but we should try to focus at least on some of them.

**Comment by beta-ziliani** [comment]
An alternative could be to add each as opt-in, and at some point turn them all by default

**Comment by straight-shoota** [comment]
Considering that the 1.8 release is approaching soon and we have had no progress on any other changes, it's probably best to delay the release of formatter changes.

We should disable the changes from https://github.com/crystal-lang/crystal/pull/12951 by default for 1.8. 
They can be available behind a flag.

**Comment by beta-ziliani** [comment]
Let's add a `preview` flag to stuff in there all the stuff coming in a future release

**Comment by HertzDevil** [comment]
The formatter does not actually support `-D` flags right now. If we want to hide things behind a flag without "implementing" flags themselves for 1.8 it seems our only option is an environment variable

**Comment by straight-shoota** [comment]
Yeah, I was planning to first implement it with an internal flag only. This allows us to keep the specs running. We can still consider exposing it in the CLI, but that would be a separate step.

**Comment by HertzDevil** [comment]
If we have multiple formatter changes but they are still not enough, do we have just a single flag for all of them, or individual flags?

**Comment by beta-ziliani** [comment]
I was thinking in just one flag

**Comment by straight-shoota** [comment]
Specific flags have some use internally as it clearly denotes what parts of the code belong to that feature. That's how https://github.com/crystal-lang/crystal/pull/13215 works.
If we expose preview mode in the CLI, I would only use a single flag to enables all available internal features.

**Comment by straight-shoota** [comment]
In https://github.com/crystal-lang/crystal/pull/14158#discussion_r1439854555 @Blacksmoke16 brought up that we probably shouldn't wait for *all* these issues to be fixed.
And yeah, this list is just meant to collect formatter changes for the purpose of grouping them.

I'm not sure what should drive the decision to enable all pending formatter changes.
The idea of this to avoid formatter changes in every single release, but it's not meant to hold up improvements for a long time.
#12951 for example has been merged for a year now. I think it oughta be time for it to actually roll out.
https://github.com/crystal-lang/crystal/issues/13169 is also ready. So we can take those two for now.

I wouldn't want to do this for 1.11 though because we're already in the freeze period.
But let's schedule their activation for 1.12.
